app/services/proofing/resolution/plugins/threat_metrix_plugin.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Resolution
+    module Plugins
+      class ThreatMetrixPlugin
+        def call(
+          applicant_pii:,
+          current_sp:,
+          request_ip:,
+          threatmetrix_session_id:,
+          timer:,
+          user_email:
+        )
+          unless FeatureManagement.proofing_device_profiling_collecting_enabled?
+            return threatmetrix_disabled_result
+          end
+
+          # The API call will fail without a session ID, so do not attempt to make
+          # it to avoid leaking data when not required.
+          return threatmetrix_id_missing_result if threatmetrix_session_id.blank?
+          return threatmetrix_pii_missing_result if applicant_pii.blank?
+
+          ddp_pii = applicant_pii.dup
+          ddp_pii[:threatmetrix_session_id] = threatmetrix_session_id
+          ddp_pii[:email] = user_email
+          ddp_pii[:request_ip] = request_ip
```

Comment:
I know this is a lift-n-ship but this could all be one method call
```suggestion
          ddp_pii = applicant_pii.merge(
            threatmetrix_session_id: threatmetrix_session_id,
            email: user_email,
            request_id: request_ip,
          )
```